src/main/java/edu/indiana/dlib/amppd/service/impl/BatchServiceImpl.java

```diff
@@ -160,6 +161,11 @@ private Primaryfile createPrimaryfile(Collection batchfileCollection, Item item,
 			{
 				primaryFile.setItem(item);
 				Set<Primaryfile> primaryfilesSet = item.getPrimaryfiles();
+				log.info("primaryfilesSet :::"+primaryfilesSet);
+				if(primaryfilesSet == null)
+				{
+					primaryfilesSet = new HashSet<Primaryfile>();
+				}
```

Comment:
if primaryfilesSet is null you will get NPE on lie 164. you may want to move the check before log. Alternatively, you can also return new empty set in the item.getPrimaryfiles - in fact, did you try that method, i wonder what it returns when no primaryfile exists for the item. My impression is that in general spring or hibernate tends to return empty list/set instead of null.

Reply:
I had to remove that log statement. It was added for debugging and i forgot to take it off.

Reply:
fixed